src/sql/platform/telemetry/telemetry.contribution.ts

```diff
@@ -13,42 +13,124 @@ import { IStorageService, StorageScope } from 'vs/platform/storage/common/storag
-		const weeklyLastUseDate = Date.parse(storageService.get('telemetry.weeklyLastUseDate', StorageScope.GLOBAL, '0'));
-		const monthlyLastUseDate = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const dailyLastUseDate: number = Date.parse(storageService.get('telemetry.dailyLastUseDate', StorageScope.GLOBAL, '0'));
+		const weeklyLastUseDate: number  = Date.parse(storageService.get('telemetry.weeklyLastUseDate', StorageScope.GLOBAL, '0'));
+		const monthlyLastUseDate: number  = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const firstTimeUser: boolean = dailyLastUseDate === Date.parse('0');
 
-		let today = new Date().toUTCString();
+		let today: Date = new Date();
+		let todayString: string = new Date().toUTCString();
 
 		// daily user event
-		if (this.diffInDays(Date.parse(today), dailyLastUseDate) >= 1) {
+		if (this.diffInDays(Date.parse(todayString), dailyLastUseDate) >= 1) {
 			// daily first use
 			telemetryService.publicLog('telemetry.dailyFirstUse', { dailyFirstUse: true });
-			storageService.store('telemetry.dailyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.dailyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
 		// weekly user event
-		if (this.diffInDays(Date.parse(today), weeklyLastUseDate) >= 7) {
+		if (this.diffInDays(Date.parse(todayString), weeklyLastUseDate) >= 7) {
 			// weekly first use
 			telemetryService.publicLog('telemetry.weeklyFirstUse', { weeklyFirstUse: true });
-			storageService.store('telemetry.weeklyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.weeklyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
-		// monthly user events
-		if (this.diffInDays(Date.parse(today), monthlyLastUseDate) >= 30) {
+
+		/* send monthly uses at the end of month */
+		const monthlyUseCount: number = storageService.getNumber('telemetry.monthlyUseCount', StorageScope.GLOBAL, 0);
+		if (this.diffInDays(Date.parse(todayString), monthlyLastUseDate) >= 30) {
 			telemetryService.publicLog('telemetry.monthlyUse', { monthlyFirstUse: true });
-			storageService.store('telemetry.monthlyLastUseDate', today, StorageScope.GLOBAL);
-		}
+			// the month changed, so send the user usage type event based on monthly count for last month
+			// and reset the count for this month
+			let lastMonthDate = new Date(monthlyLastUseDate);
+			this.sendUsageEvent(monthlyUseCount, lastMonthDate);
+
+			const wasActiveLastMonth: boolean = storageService.getBoolean('telemetry.wasActiveLastMonth', StorageScope.GLOBAL, false);
+			const isActiveThisMonth: boolean = storageService.getNumber('telemetry.monthlyUseCount', StorageScope.GLOBAL, 0) >= 1;
+			const isContinuing: boolean = wasActiveLastMonth && isActiveThisMonth;
+			const isReturning: boolean = !wasActiveLastMonth && isActiveThisMonth;
+
+			if (firstTimeUser) {
```

Comment:
nit: To reduce duplicated calls I'd just calculate the userGrowthType separately and then just have a single call to publicLog (reduces chance of errors if you mistype in one of them)